app/controllers/user_controller.rb

```diff
@@ -0,0 +1,44 @@
+post '/users/signup' do
+  @user = User.new(:name => params[:user][:name], :password => params[:user][:password])
+  if @user.save
+    erb :'/users/new'
+  else
+    "Cannot sign up, username already exists"
```

Comment:
Let's get accurate error messaging here, since the reasons for user create failure include possibilities besides username already existing (e.g. blank password)

Reply:
True. I added more cases now. Not sure how to specifically check that the username already exists.